core-java-modules/core-java-collections-list-6/src/main/java/com/baeldung/sublistindexoutofboundsexception/SubListUtils.java

```diff
@@ -0,0 +1,24 @@
+package com.baeldung.sublistindexoutofboundsexception;
+
+import java.util.Collections;
+import java.util.List;
+
+public class SubListUtils {
+
+    public static List<String> safeSubList(List<String> myList, int fromIndex, int toIndex) {
+        if (myList == null) {
+            return null;
+        }
+
+        int size = myList.size();
+        if (fromIndex >= size || toIndex <= 0 || fromIndex >= toIndex) {
+            return Collections.emptyList();
+        }
+
+        fromIndex = Math.max(0, fromIndex);
+        toIndex = Math.min(size, toIndex);
+
+        return myList.subList(fromIndex, toIndex);
+    }
```

Comment:
A simpler version of this method:
```suggestion
public static List<String> safeSubList(List<String> myList, int fromIndex, int toIndex) {
  if (myList == null || fromIndex >= myList.size() || toIndex <= 0 || fromIndex >= toIndex) {
    return Collections.emptyList();
  }
  return myList.subList(Math.max(0, fromIndex), Math.min(myList.size(), toIndex));
}
```